webclient/webclient/src/main/java/io/helidon/webclient/WebClientResponseImpl.java

```diff
@@ -72,11 +75,19 @@ public static Builder builder() {
         return version;
     }
 
+    @Override
+    public URI lastEndpointURI() {
+        return lastEndpointUri;
+    }
+
     @Override
     public CompletionStage<Void> close() {
+        if (responseCloser.isClosed()) {
+            return CompletableFuture.completedFuture(null);
+        }
         CompletableFuture<Void> toReturn = new CompletableFuture<>();
         responseCloser.close().addListener(future -> {
-            LOGGER.finest("Response has been closed.");
+            LOGGER.finest("Response from " + lastEndpointUri + " has been closed.");
```

Comment:
This should use a lambda to log
`LOGGER.finest(() -> "...")`

Reply:
Thanks.